### PR TITLE
Add BeagleBone-DMX driver

### PR DESCRIPTION
**Pull request by xuio**
I added compatibility with https://github.com/boxysean/beaglebone-DMX (Beaglebone-PRU-DMX driver).
It works well, tested on my BeagleBone Black

### CODE REVIEW

File: drivers/bbdmx.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	} else {
+		self.port = options.port || 9930;
+	}
+	self.dev = dgram.createSocket('udp4');
+	self.sleepTime = 24;
+	self.start();
+}
+
+BBDMX.prototype.send_universe = function() {
+	let messageBuffer = new Buffer(`${this.universe.length} `);
```
Comment (wiedi) [inline drivers/bbdmx.js]: could you adjust this to `new Buffer(this.universe.length + ' ')` so it'll still work with oder node versions?

File: drivers/bbdmx.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+	for (let i = 0; i < this.universe.length; i++) {
+		const channel = new Buffer(`${this.universe[i]} `);
```
Comment (wiedi) [inline drivers/bbdmx.js]: same here: `new Buffer(this.universe[i] + ' ')` - should work just as well, right?

**Comment by wiedi** [comment]
Also, to quote the protocol description:
`A single space is expected between the values, and there should be no space after the last value.`

Currently there will be a space in the end, wouldn't it be better to conform to this?

**Comment by xuio** [comment]
Thanks for noticing. You're totally right, I just pushed a fix.

Should I also remove the `let` and `const` variable prefixes for backwards compatibility?
iirc they can be used in strickt mode with older node versions.

**Comment by wiedi** [comment]
I think those are fine, at least with 0.12

**Pull request merged**